### ISSUE TITLE
Validation fails when all controls on the form are disabled and all have values

### ISSUE THREAD

**Issue by zzhimin** [ngx-formly/ngx-formly#4013]
**Description**
Validation fails when all controls on the form are disabled and all have values

**Minimal Reproduction**
1. Open official example: https://formly.dev/docs/examples/validation/toggle-required；
2. Click Edit in StackBlitz;
3. Replace fields：[
    {
      key: 'text1',
      type: 'input',
      defaultValue: 'value1',
      templateOptions: {
        label: 'Moehahah',
        placeholder: 'Formly is terrific!',
        required: false,
        disabled: true,
      },
    },
    {
      key: 'text2',
      type: 'input',
      defaultValue: 'value2',
      templateOptions: {
        label: 'Moehahah',
        placeholder: 'Formly is terrific!',
        required: false,
        disabled: true,
      },
    },
  ]；

![image](https://github.com/user-attachments/assets/0271ea42-0697-417d-873c-34152a2dd37f)

**Comment by lattaai1** [comment]
Hello, I've fixed this issue!

This is what I've done:
I have identified the issue and implemented a fix for the validation problem when all controls on the form are disabled and have values. The problem was in the `FieldFormExtension` class, specifically in the `setValidators` method. I've updated the logic to ensure that validators are still applied even when the control is disabled. Here's a summary of the changes:

1. Modified the `setValidators` method in `src/core/src/lib/extensions/field-form/field-form.ts`.
2. Updated the logic to apply validators regardless of the control's disabled state.
3. Ensured that the form's validity is updated correctly for disabled controls.

These changes should resolve the issue while maintaining the existing functionality for enabled controls.

I also created pull request: https://github.com/ngx-formly/ngx-formly/pull/4014


> [!CAUTION]  
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission
